proposals/required-members.md

```diff
@@ -263,42 +263,131 @@ init_argument_initializer
 A type with a parameterless constructor that advertises a _contract_ is not allowed to be substituted for a type parameter constrained to `new()`, as there is no way
 for the generic instantiation to ensure that the requirements are satisfied.
 
+### Overriding, Hiding, and Inheriting
+
+It is an error to mark a member required if the member is not at least as visible as the containing member's type. This means the following cases are not allowed:
+
+```cs
+interface I
+{
+    int Prop1 { get; }
+}
+public class Base
+{
+    public virtual int Prop2 { get; }
+
+    protected required int _field; // Error: _field is not at least as visible as Base. Open question below about the protected constructor scenario
+    protected Base() { }
+}
+public class Derived : Base, I
+{
+    required int I.Prop1 { get; } // Error: explicit interface implementions cannot be required as they cannot be set in an object initializer
+
+    public required override int Prop2 { get; } // Error: this property is hidden by Derived.Prop2 and cannot be set in an object initializer
+    public new int Prop2 { get; }
+}
+```
+
+It is an error to hide a `required` member, as that member can no longer be set by a consumer.
```

Comment:
nit: I don't know if Mads or someone else has created a proposal for addressing members of specific types (something like `x.(Type.Method)(argument)`). If there's one, let's add a note there about this scenario.

Reply:
@MadsTorgersen?